lms/static/scripts/frontend_apps/components/test/GroupConfigSelector-test.js

```diff
@@ -193,22 +197,46 @@ describe('GroupConfigSelector', () => {
     assert.equal(options.length, fakeGroupSets.length);
   });
 
-  it('hides authorization prompt if checkbox is de-selected', async () => {
-    fakeAPICall
-      .withArgs(groupSetsAPIRequest)
-      .rejects(new Error('Authorization failed'));
+  it('shows errors in a modal with a retry button', async () => {
+    fakeIsAuthorizationError.returns(false);
+    fakeAPICall.withArgs(groupSetsAPIRequest).rejects(new Error('Some error'));
+    const wrapper = createComponent({
+      groupConfig: { useGroupSet: true, groupSet: null },
+    });
+
+    const authModal = await waitForElement(wrapper, 'ErrorModal');
+
+    fakeAPICall.withArgs(groupSetsAPIRequest).resolves(fakeGroupSets);
+    authModal.prop('onRetry')();
```

Comment:
The `onRetry` call should be wrapped in `act` to flush any effects/state changes immediately, though `waitForElement` will eventually see them either way.